src/backend/distributed/commands/multi_copy.c

```diff
@@ -2743,6 +2755,60 @@ FinishLocalCopy(CitusCopyDestReceiver *copyDest)
 }
 
 
+/*
+ * StartLocalColocatedIntermediateFile creates a co-located file for the given
+ * shard, and appends the binary headers if needed.
+ */
+static void
+StartLocalColocatedIntermediateFile(CitusCopyDestReceiver *copyDest,
+									CopyShardState *shardState)
+{
+	/* make sure the directory exists */
+	CreateIntermediateResultsDirectory();
+
+	const int fileFlags = (O_APPEND | O_CREAT | O_RDWR | O_TRUNC | PG_BINARY);
+	const int fileMode = (S_IRUSR | S_IWUSR);
+
+	StringInfo filePath = makeStringInfo();
+	appendStringInfo(filePath, "%s_%ld", copyDest->colocatedIntermediateResultIdPrefix,
+					 shardState->shardId);
+
+	const char *fileName = QueryResultFileName(filePath->data);
+	shardState->fileDest =
+		FileCompatFromFileStart(FileOpenForTransmit(fileName, fileFlags, fileMode));
+
+	CopyOutState localFileCopyOutState = shardState->copyOutState;
+	bool isBinaryCopy = localFileCopyOutState->binary;
+	if (ShouldAddBinaryHeaders(localFileCopyOutState->fe_msgbuf, isBinaryCopy))
```

Comment:
it is enough to use `isBinaryCopy ` flag, `ShouldAddBinaryHeaders ` is not necessary and confusing.